animation_nodes/algorithms/mesh/triangulate_mesh.pyx

```diff
@@ -0,0 +1,316 @@
+    cdef unsigned int *oldPolyStarts = polygons.polyStarts.data
+
+    cdef PolygonIndicesList newPolygons = PolygonIndicesList(
+                                          indicesAmount = 3 * triAmount,
+                                          polygonAmount = triAmount)
+    cdef unsigned int *newIndices = newPolygons.indices.data
+    cdef unsigned int *newPolyStarts = newPolygons.polyStarts.data
+    cdef unsigned int *newPolyLengths = newPolygons.polyLengths.data
+
+    cdef Py_ssize_t j, index, triangleIndex, start
+
+    index = 0
+    triangleIndex = 0
+    for i in range(polygonAmount):
+        start = oldPolyStarts[i]
+        for j in range(oldPolyLengths[i] - 2):
+            newPolyStarts[triangleIndex] = index
+            newPolyLengths[triangleIndex] = 3
+            triangleIndex += 1
+
+            newIndices[index] = oldIndices[start]
+            newIndices[index + 1] = oldIndices[start + 1 + j]
+            newIndices[index + 2] = oldIndices[start + 2 + j]
+            index += 3
+
+    return newPolygons
+
+@cython.cdivision(True)
+def triangulatePolygonsUsingEarClipMethod(Vector3DList vertices, PolygonIndicesList polygons):
+    cdef unsigned int *oldPolyLengths = polygons.polyLengths.data
+    cdef Py_ssize_t polygonAmount = polygons.getLength()
+    cdef Py_ssize_t i, triAmount, polyLength
+
+    # Total number of triangle polygons.
+    triAmount = 0
+    for i in range(polygonAmount):
+        polyLength = oldPolyLengths[i]
+        triAmount += polyLength - 2
+
+    cdef unsigned int *oldIndices = polygons.indices.data
+    cdef unsigned int *oldPolyStarts = polygons.polyStarts.data
+
+    cdef PolygonIndicesList newPolygons = PolygonIndicesList(
+                                          indicesAmount = 3 * triAmount,
+                                          polygonAmount = triAmount)
+    cdef unsigned int *newIndices = newPolygons.indices.data
+    cdef unsigned int *newPolyStarts = newPolygons.polyStarts.data
+    cdef unsigned int *newPolyLengths = newPolygons.polyLengths.data
+
+    cdef float angle
+    cdef Vector3 v1, v2, v3
+    cdef Vertex* verticesData
+    cdef Py_ssize_t polyStart, triangleIndex, polyIndex
+    cdef Py_ssize_t j, index, previousIndex, earIndex, nextIndex
+
+    polyIndex = 0
+    triangleIndex = 0
+    for i in range(polygonAmount):
+        polyStart = oldPolyStarts[i]
+        polyLength = oldPolyLengths[i]
+
+        # Initialization of polygon's vertices.
+        verticesData = <Vertex*>PyMem_Malloc(polyLength * sizeof(Vertex))
+        projectPolygonVertices(vertices, polyStart, polyLength, oldIndices, verticesData)
+
+        # Check polygon polarity.
+        if not polyCounterClockwise(verticesData):
+            for j in range(polyLength):
+                previousIndex = verticesData[j].previous
+                nextIndex = verticesData[j].next
+                verticesData[j].previous = nextIndex
+                verticesData[j].next = previousIndex
```

Comment:
Put those in a function, just to make it clear what is happening.

Reply:
It is added into `makeSurePolygonIsCounterClockwise(verticesData)` function.